service/storage_service.cc

```diff
@@ -369,7 +369,8 @@ void storage_service::prepare_to_join(
     slogger.info("Starting up server gossip");
 
     auto generation_number = db::system_keyspace::increment_and_get_generation().get0();
-    _gossiper.start_gossiping(generation_number, app_states, gms::bind_messaging_port(bool(do_bind))).get();
+    auto advertise = replacing_a_node_with_same_ip ? gms::advertise_myself::no : gms::advertise_myself::yes;
```

Comment:
I think that `advertise` can be simply constructed as `gms::advertise_myself(!replacing_a_node_with_same_ip)`

Reply:
The following might work too:
```
gms::advertise_myself advertise = !replacing_a_node_with_same_ip;
```

Reply:
Yes.